gradle.properties

```diff
@@ -0,0 +1,4 @@
+org.gradle.caching = true
```

Comment:
Regarding `org.gradle.caching`, `org.gradle.parallel` and `org.gradle.configureondemand`, have you measured performance before and after enabling these properties? If yes, what are the numbers?

Reply:
The [Build Time](https://github.com/JetBrains/jediterm/actions/runs/1436912709) is 2m49s, which is better than the most recent build[Build Time](https://github.com/JetBrains/jediterm/actions/runs/1436564092)

Our strategies didn't show a huge improvement, and in fact the project had about 320 source files.
If the project continues to grow, it may become a large source data project, at which point its build time will increase significantly, and our strategy will make a significant difference.

In addition,`org.gradle.configureondemand=true` works when we attempts to configure only projects that are relevant for requested tasks
`org.gradle.caching=true` helps us build faster in incremental development

Reply:
Thanks for the info. As I can see [`org.gradle.configureondemand`](https://docs.gradle.org/current/userguide/multi_project_configuration_and_execution.html#sec:configuration_on_demand) feature is incubating. So, probably, it'd be better to refrain from enabling it. Regarding `org.gradle.caching` and `org.gradle.parallel`, are these options enabled by default? If no, why not? Are they stable enough to be enabled? Asking, because enabling these options gives some performance gain, but it is not huge for this project. Having stable builds is more important. Would be great to ensure that enabling these options won't result in broken builds.

Reply:
Thanks for reply! I fully understand your concern.

Gradle have made some configuration enabled by default, for instance, **_Gradle daemon_** enabled by default since gradle 3.0 and  **_File system watching_** enabled by default since gradle 7.0

Gradle documentation does not explain why other configurations are not enabled by default, since there are not many configurations that are enabled by default.

`org.gradle.caching` works when this build can use the previous build cache, if not , it will work fine and will not take advantage of caching.  So it would not bring unstable things.

As for `org.gradle.parallel`,  [doc](https://docs.gradle.org/current/userguide/multi_project_configuration_and_execution.html#sec:parallel_execution) says :"parallel execution does not strictly require decoupling at configuration time". and "Gradle attempts to choose the right number based on available CPU cores"

since `Configuration on demand`  is incubating, to make this project stable, we can remove it.
I sincerely recommend you to use the remaining two configurations 

Have a nice day!

Reply:
Thanks for the information. I'll merge it a bit later, after IntelliJ IDA 2021.3 release.